<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>584</width>
    <height>561</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>584</width>
    <height>561</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>584</width>
    <height>561</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Поиск глобального минимума</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>330</y>
      <width>271</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Имитация отжига</string>
    </property>
    <widget class="QCheckBox" name="checkHJIO">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>231</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Локальная оптимизация Хука-Дживса</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Tmax=</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="tbTmax">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>20</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>10000,01</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>18</x>
       <y>140</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Время работы</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>60</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>e = </string>
     </property>
    </widget>
    <widget class="QPushButton" name="bIO">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>231</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Имитация отжига</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="tbr">
     <property name="geometry">
      <rect>
       <x>62</x>
       <y>60</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0,1</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>r =</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="tbIOtime">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>140</y>
       <width>151</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>20</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>L = </string>
     </property>
    </widget>
    <widget class="QLineEdit" name="tbEps">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>60</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0,1</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="tbL">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>20</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>1000</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>330</y>
      <width>271</width>
      <height>201</height>
     </rect>
    </property>
    <property name="title">
     <string>Монте-Карло</string>
    </property>
    <widget class="QLineEdit" name="tbSizeTest">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>20</y>
       <width>111</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>5000</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>140</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Время работы</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="tbMCtime">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>140</y>
       <width>171</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>12</x>
       <y>20</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Количество испытаний</string>
     </property>
    </widget>
    <widget class="QPushButton" name="bMC">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>251</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>Монте-Карло</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="checkHJMK">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>231</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Локальная оптимизация Хука-Дживса</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>10</y>
      <width>271</width>
      <height>311</height>
     </rect>
    </property>
    <property name="title">
     <string>Выходные данные</string>
    </property>
    <widget class="QTableWidget" name="dgvOut">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>241</width>
       <height>281</height>
      </rect>
     </property>
     <row>
      <property name="text">
       <string>f(x)</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>x1</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>x2</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>x3</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>x4</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>Имитация обжига</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Монте-Карло</string>
      </property>
     </column>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>261</width>
      <height>311</height>
     </rect>
    </property>
    <property name="title">
     <string>Входные данные</string>
    </property>
    <widget class="QTableWidget" name="dgvXLimInput">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>241</width>
       <height>221</height>
      </rect>
     </property>
     <row>
      <property name="text">
       <string/>
      </property>
     </row>
     <row>
      <property name="text">
       <string/>
      </property>
     </row>
     <row>
      <property name="text">
       <string/>
      </property>
     </row>
     <row>
      <property name="text">
       <string/>
      </property>
     </row>
     <column>
      <property name="text">
       <string>Имя переменной</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Начало</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Конец</string>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string>x1</string>
      </property>
     </item>
     <item row="0" column="1">
      <property name="text">
       <string>-10</string>
      </property>
     </item>
     <item row="0" column="2">
      <property name="text">
       <string>10</string>
      </property>
     </item>
     <item row="1" column="0">
      <property name="text">
       <string>x2</string>
      </property>
     </item>
     <item row="1" column="1">
      <property name="text">
       <string>-10</string>
      </property>
     </item>
     <item row="1" column="2">
      <property name="text">
       <string>10</string>
      </property>
     </item>
     <item row="2" column="0">
      <property name="text">
       <string>x3</string>
      </property>
     </item>
     <item row="2" column="1">
      <property name="text">
       <string>-10</string>
      </property>
     </item>
     <item row="2" column="2">
      <property name="text">
       <string>10</string>
      </property>
     </item>
     <item row="3" column="0">
      <property name="text">
       <string>x4</string>
      </property>
     </item>
     <item row="3" column="1">
      <property name="text">
       <string>-10</string>
      </property>
     </item>
     <item row="3" column="2">
      <property name="text">
       <string>10</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>22</y>
       <width>47</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Функции</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>131</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Количество переменных</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="cbFunction">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>20</y>
       <width>191</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>100*(x2-x2^2)^2+(1-x1)^2+90*(x4-x3^2)^2+(1-x3)^2+10.1*((x2-1)^2+(x4-1)^2)+19.8*(x2-1)*(x4-1)</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="tbPer">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>50</y>
       <width>91</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>4</number>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>584</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="action_4"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>О программе</string>
    </property>
    <addaction name="action_Qt"/>
    <addaction name="separator"/>
    <addaction name="action_Help"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <action name="action_Qt">
   <property name="text">
    <string>О Qt</string>
   </property>
  </action>
  <action name="action_2">
   <property name="text">
    <string>Сохранить в файл</string>
   </property>
  </action>
  <action name="action_4">
   <property name="text">
    <string>Выход</string>
   </property>
  </action>
  <action name="action_Help">
   <property name="text">
    <string>Доступные функции</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>action_4</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>291</x>
     <y>279</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
